File: CHPT/source/trunk/supp_app/SspDispatchService/src/com/ctfo/storage/model/permission/SysFunction.java
package com.ctfo.storage.model.permission;

import java.io.Serializable;

/**
 * 
 * 
 * <p>
 * ----------------------------------------------------------------------------- <br>
 * 工程名 ： SspDispatchService <br>
 * 功能： 功能菜单表<br>
 * 描述： 功能菜单表<br>
 * 授权 : (C) Copyright (c) 2011 <br>
 * 公司 : 北京中交慧联信息科技有限公司 <br>
 * ----------------------------------------------------------------------------- <br>
 * 修改历史 <br>
 * <table width="432" border="1">
 * <tr>
 * <td>版本</td>
 * <td>时间</td>
 * <td>作者</td>
 * <td>改变</td>
 * </tr>
 * <tr>
 * <td>1.0</td>
 * <td>2014-11-4</td>
 * <td>xuehui</td>
 * <td>创建</td>
 * </tr>
 * </table>
 * <br>
 * <font color="#FF0000">注意: 本内容仅限于[北京中交慧联信息科技有限公司]内部使用，禁止转发</font> <br>
 * 
 * @version 1.0
 * 
 * @author xuehui
 * @since JDK1.6
 */
public class SysFunction implements Serializable {

	/** */
	private static final long serialVersionUID = 1196651202199647894L;

	/** 自定义编码 */
	private String funId;

	/** 序号 */
	private Integer num;

	/** 功能名称 */
	private String funName;

	/** 菜单英文名 */
	private String funEname;

	/** 功能对应地址 */
	private String funUri;

	/** 对应的图标文件地址 */
	private String funImg;

	/** 上一级功能编号 */
	private String parentId;

	/** 1，服务端功能 2，客户端功能 */
	private String funCbs;

	/** 0-9 */
	private Integer funLevel;

	/** 执行方法 */
	private String funIdx;

	/** 0：未启用1：启用 */
	private Integer funFlag;

	/** 1：可运行功能0：不可 运行功能，属于菜单 */
	private Integer funRun;

	/** 按钮--浏览 0无，1有 */
	private String buttonBrowse;

	/** 按钮--保存 0无，1有 */
	private String buttonSave;

	/** 按钮--新增 0无，1有 */
	private String buttonAdd;

	/** 按钮--编辑 0无，1有 */
	private String buttonEdit;

	/** 按钮--复制 0无，1有 */
	private String buttonCopy;

	/** 按钮--删除 0无，1有 */
	private String buttonDelete;

	/** 按钮--作废 0无，1有 */
	private String buttonCancel;

	/** 按钮--激活 0无，1有 */
	private String buttonActivate;

	/** 按钮--提交 0无，1有 */
	private String buttonSubmit;

	/** 按钮--导入 0无，1有 */
	private String buttonImport;

	/** 按钮--导出 0无，1有 */
	private String buttonExport;

	/** 按钮--打印 0无，1有 */
	private String buttonPrint;

	/** 按钮--操作记录 0无，1有 */
	private String buttonOperationRecord;

	/** 按钮--派工 0无，1有 */
	private String buttonDispatching;

	/** 1:有效 0:无效 默认为1 */
	private String enableFlag;

	/** 备注 */
	private String remark;

	/** 创建人，关联人员表 */
	private String createBy;

	/** 创建时间 */
	private Long createTime;

	/** 最后修改人，关联人员表 */
	private String updateBy;

	/** 最后修改时间 */
	private Long updateTime;

	/** 服务站id，云平台用 */
	private String serStationId;

	/** 帐套id，云平台用 */
	private String setBookId;

	private String buttonSync;

	private String buttonBalance;

	private String buttonRevoke;

	private String buttonCommit;

	private String buttonVerify;

	private String buttonStatus;

	private String buttonConfirm;

	public String getSerStationId() {
		return serStationId;
	}

	public void setSerStationId(String serStationId) {
		this.serStationId = serStationId;
	}

	public String getSetBookId() {
		return setBookId;
	}

	public void setSetBookId(String setBookId) {
		this.setBookId = setBookId;
	}

	public String getFunId() {
		return funId;
	}

	public void setFunId(String funId) {
		this.funId = funId;
	}

	public Integer getNum() {
		return num;
	}

	public void setNum(Integer num) {
		this.num = num;
	}

	public String getFunName() {
		return funName;
	}

	public void setFunName(String funName) {
		this.funName = funName;
	}

	public String getFunEname() {
		return funEname;
	}

	public void setFunEname(String funEname) {
		this.funEname = funEname;
	}

	public String getFunUri() {
		return funUri;
	}

	public void setFunUri(String funUri) {
		this.funUri = funUri;
	}

	public String getFunImg() {
		return funImg;
	}

	public void setFunImg(String funImg) {
		this.funImg = funImg;
	}

	public String getParentId() {
		return parentId;
	}

	public void setParentId(String parentId) {
		this.parentId = parentId;
	}

	public String getFunCbs() {
		return funCbs;
	}

	public void setFunCbs(String funCbs) {
		this.funCbs = funCbs;
	}

	public Integer getFunLevel() {
		return funLevel;
	}

	public void setFunLevel(Integer funLevel) {
		this.funLevel = funLevel;
	}

	public String getFunIdx() {
		return funIdx;
	}

	public void setFunIdx(String funIdx) {
		this.funIdx = funIdx;
	}

	public Integer getFunFlag() {
		return funFlag;
	}

	public void setFunFlag(Integer funFlag) {
		this.funFlag = funFlag;
	}

	public Integer getFunRun() {
		return funRun;
	}

	public void setFunRun(Integer funRun) {
		this.funRun = funRun;
	}

	public String getButtonBrowse() {
		return buttonBrowse;
	}

	public void setButtonBrowse(String buttonBrowse) {
		this.buttonBrowse = buttonBrowse;
	}

	public String getButtonSave() {
		return buttonSave;
	}

	public void setButtonSave(String buttonSave) {
		this.buttonSave = buttonSave;
	}

	public String getButtonAdd() {
		return buttonAdd;
	}

	public void setButtonAdd(String buttonAdd) {
		this.buttonAdd = buttonAdd;
	}

	public String getButtonEdit() {
		return buttonEdit;
	}

	public void setButtonEdit(String buttonEdit) {
		this.buttonEdit = buttonEdit;
	}

	public String getButtonCopy() {
		return buttonCopy;
	}

	public void setButtonCopy(String buttonCopy) {
		this.buttonCopy = buttonCopy;
	}

	public String getButtonDelete() {
		return buttonDelete;
	}

	public void setButtonDelete(String buttonDelete) {
		this.buttonDelete = buttonDelete;
	}

	public String getButtonCancel() {
		return buttonCancel;
	}

	public void setButtonCancel(String buttonCancel) {
		this.buttonCancel = buttonCancel;
	}

	public String getButtonActivate() {
		return buttonActivate;
	}

	public void setButtonActivate(String buttonActivate) {
		this.buttonActivate = buttonActivate;
	}

	public String getButtonSubmit() {
		return buttonSubmit;
	}

	public void setButtonSubmit(String buttonSubmit) {
		this.buttonSubmit = buttonSubmit;
	}

	public String getButtonImport() {
		return buttonImport;
	}

	public void setButtonImport(String buttonImport) {
		this.buttonImport = buttonImport;
	}

	public String getButtonExport() {
		return buttonExport;
	}

	public void setButtonExport(String buttonExport) {
		this.buttonExport = buttonExport;
	}

	public String getButtonPrint() {
		return buttonPrint;
	}

	public void setButtonPrint(String buttonPrint) {
		this.buttonPrint = buttonPrint;
	}

	public String getButtonOperationRecord() {
		return buttonOperationRecord;
	}

	public void setButtonOperationRecord(String buttonOperationRecord) {
		this.buttonOperationRecord = buttonOperationRecord;
	}

	public String getButtonDispatching() {
		return buttonDispatching;
	}

	public void setButtonDispatching(String buttonDispatching) {
		this.buttonDispatching = buttonDispatching;
	}

	public String getEnableFlag() {
		return enableFlag;
	}

	public void setEnableFlag(String enableFlag) {
		this.enableFlag = enableFlag;
	}

	public String getRemark() {
		return remark;
	}

	public void setRemark(String remark) {
		this.remark = remark;
	}

	public String getCreateBy() {
		return createBy;
	}

	public void setCreateBy(String createBy) {
		this.createBy = createBy;
	}

	public Long getCreateTime() {
		return createTime;
	}

	public void setCreateTime(Long createTime) {
		this.createTime = createTime;
	}

	public String getUpdateBy() {
		return updateBy;
	}

	public void setUpdateBy(String updateBy) {
		this.updateBy = updateBy;
	}

	public Long getUpdateTime() {
		return updateTime;
	}

	public void setUpdateTime(Long updateTime) {
		this.updateTime = updateTime;
	}

	public String getButtonSync() {
		return buttonSync;
	}

	public void setButtonSync(String buttonSync) {
		this.buttonSync = buttonSync;
	}

	public String getButtonBalance() {
		return buttonBalance;
	}

	public void setButtonBalance(String buttonBalance) {
		this.buttonBalance = buttonBalance;
	}

	public String getButtonRevoke() {
		return buttonRevoke;
	}

	public void setButtonRevoke(String buttonRevoke) {
		this.buttonRevoke = buttonRevoke;
	}

	public String getButtonCommit() {
		return buttonCommit;
	}

	public void setButtonCommit(String buttonCommit) {
		this.buttonCommit = buttonCommit;
	}

	public String getButtonVerify() {
		return buttonVerify;
	}

	public void setButtonVerify(String buttonVerify) {
		this.buttonVerify = buttonVerify;
	}

	public String getButtonStatus() {
		return buttonStatus;
	}

	public void setButtonStatus(String buttonStatus) {
		this.buttonStatus = buttonStatus;
	}

	public String getButtonConfirm() {
		return buttonConfirm;
	}

	public void setButtonConfirm(String buttonConfirm) {
		this.buttonConfirm = buttonConfirm;
	}

}

File: CHPT/source/tags/V0.4.4.1/client_app/Model/sys_function.cs
using System;
namespace Model
{
	/// <summary>
	/// 功能菜单
	/// </summary>
	[Serializable]
	public partial class sys_function
	{
		public sys_function()
		{}
		#region Model
		private string _fun_id;
		private int? _num;
		private string _fun_name;
		private string _fun_ename;
		private string _fun_uri;
		private string _fun_img;
		private string _parent_id;
		private string _fun_cbs;
		private int? _fun_level;
		private string _fun_idx;
		private int? _fun_flag;
		private int? _fun_run;
		private string _enable_flag;
		private string _remark;
		private string _create_by;
		private long _create_time;
		private string _update_by;
		private long? _update_time;
		/// <summary>
		/// 自定义编码
		/// </summary>
		public string fun_id
		{
			set{ _fun_id=value;}
			get{return _fun_id;}
		}
		/// <summary>
		/// 序号
		/// </summary>
		public int? num
		{
			set{ _num=value;}
			get{return _num;}
		}
		/// <summary>
		/// 功能名称
		/// </summary>
		public string fun_name
		{
			set{ _fun_name=value;}
			get{return _fun_name;}
		}
		/// <summary>
		/// 菜单英文名
		/// </summary>
		public string fun_ename
		{
			set{ _fun_ename=value;}
			get{return _fun_ename;}
		}
		/// <summary>
		/// 功能对应地址
		/// </summary>
		public string fun_uri
		{
			set{ _fun_uri=value;}
			get{return _fun_uri;}
		}
		/// <summary>
		/// 对应的图标文件地址
		/// </summary>
		public string fun_img
		{
			set{ _fun_img=value;}
			get{return _fun_img;}
		}
		/// <summary>
		/// 上一级功能编号
		/// </summary>
		public string parent_id
		{
			set{ _parent_id=value;}
			get{return _parent_id;}
		}
		/// <summary>
		/// 1，服务端功能   2，客户端功能
		/// </summary>
		public string fun_cbs
		{
			set{ _fun_cbs=value;}
			get{return _fun_cbs;}
		}
		/// <summary>
		/// 0-9
		/// </summary>
		public int? fun_level
		{
			set{ _fun_level=value;}
			get{return _fun_level;}
		}
		/// <summary>
		/// 执行方法
		/// </summary>
		public string fun_idx
		{
			set{ _fun_idx=value;}
			get{return _fun_idx;}
		}
		/// <summary>
		/// 0：未启用1：启用
		/// </summary>
		public int? fun_flag
		{
			set{ _fun_flag=value;}
			get{return _fun_flag;}
		}
		/// <summary>
		/// 1：可运行功能0：不可\r\n运行功能，属于菜单
		/// </summary>
		public int? fun_run
		{
			set{ _fun_run=value;}
			get{return _fun_run;}
		}
		/// <summary>
		/// 1:有效 0:无效 默认为1
		/// </summary>
		public string enable_flag
		{
			set{ _enable_flag=value;}
			get{return _enable_flag;}
		}
		/// <summary>
		/// 备注
		/// </summary>
		public string remark
		{
			set{ _remark=value;}
			get{return _remark;}
		}
		/// <summary>
		/// 创建人，关联人员表
		/// </summary>
		public string create_by
		{
			set{ _create_by=value;}
			get{return _create_by;}
		}
		/// <summary>
		/// 创建时间
		/// </summary>
		public long create_time
		{
			set{ _create_time=value;}
			get{return _create_time;}
		}
		/// <summary>
		/// 最后修改人，关联人员表
		/// </summary>
		public string update_by
		{
			set{ _update_by=value;}
			get{return _update_by;}
		}
		/// <summary>
		/// 最后修改时间
		/// </summary>
		public long? update_time
		{
			set{ _update_time=value;}
			get{return _update_time;}
		}
		#endregion Model

	}
}

